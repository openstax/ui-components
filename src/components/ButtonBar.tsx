import React from "react";
import styled from "styled-components";
import {
  buttonBarItemCss,
  buttonBarWrapperCss,
  tabBaseCss,
  tabListBaseCss,
  TabsProps,
} from "./Tabs";

type ButtonBarProps = Pick<TabsProps, "size"> & {
  children?: React.ReactNode;
};

const Wrapper = styled.div`
  ${tabListBaseCss}
  ${buttonBarWrapperCss}

  > * {
    all: unset;
    ${tabBaseCss}
    ${buttonBarItemCss}
  }
`;

export const ButtonBar = ({
  size = "medium",
  children,
  ...restProps
}: ButtonBarProps) => {
  return (
    <Wrapper size={size} {...restProps}>
      {children}
    </Wrapper>
  );
};
