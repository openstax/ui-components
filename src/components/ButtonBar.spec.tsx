import { render } from "@testing-library/react";
import { ButtonBar } from "./ButtonBar";

describe("ButtonBar", () => {
  it("renders", () => {
    const { asFragment } = render(
      <ButtonBar>
        <a>One</a>
        <a>Two</a>
      </ButtonBar>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders small size", () => {
    const { asFragment } = render(
      <ButtonBar size="small">
        <button>One</button>
        <button>Two</button>
      </ButtonBar>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders large size", () => {
    const { asFragment } = render(
      <ButtonBar size="large">
        <button>One</button>
        <button data-selected>Two</button>
      </ButtonBar>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
