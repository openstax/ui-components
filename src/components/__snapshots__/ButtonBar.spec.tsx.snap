// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonBar renders 1`] = `
<DocumentFragment>
  <div
    class="sc-gsnTZi hYiJLm"
  >
    <a>
      One
    </a>
    <a>
      Two
    </a>
  </div>
</DocumentFragment>
`;

exports[`ButtonBar renders large size 1`] = `
<DocumentFragment>
  <div
    class="sc-gsnTZi hZAvUF"
  >
    <button>
      One
    </button>
    <button
      data-selected="true"
    >
      Two
    </button>
  </div>
</DocumentFragment>
`;

exports[`ButtonBar renders small size 1`] = `
<DocumentFragment>
  <div
    class="sc-gsnTZi gFPpQi"
  >
    <button>
      One
    </button>
    <button>
      Two
    </button>
  </div>
</DocumentFragment>
`;
